typescript/cli/src/commands/registry.ts

```diff
@@ -105,52 +105,53 @@ const createAgentConfigCommand: CommandModuleWithContext<{
 }> = {
   command: 'agent-config',
   describe: 'Create a new agent config',
-  builder: {
-    chains: chainTargetsCommandOption,
-    environment: environmentCommandOption,
-    out: outputFileCommandOption(
-      './configs/agent-config.json',
-      false,
-      'The path to output an agent config JSON file.',
-    ),
-  },
-  // TODO: make chains and environment mutually exclusive, but require one of them
-  // builder: (yargs) => {
-  //   yargs
-  //     .option('chains', chainTargetsCommandOption)
-  //     .option('environment', environmentCommandOption as Options<string>)
-  //     .option(
-  //       'out',
-  //       outputFileCommandOption(
-  //         './configs/agent-config.json',
-  //         false,
-  //         'The path to output an agent config JSON file.',
-  //       ) as Options<string>,
-  //     )
-  //     .check((argv) => {
-  //       if (!argv.chains && !argv.environment) {
-  //         throw new Error(
-  //           'Either --chains or --environment must be specified.',
-  //         );
-  //       }
-  //       return true;
-  //     })
-  //     .check((argv) => {
-  //       if (argv.chains && argv.environment) {
-  //         throw new Error(
-  //           '--chains and --environment cannot be specified together.',
-  //         );
-  //       }
-  //       return true;
-  //     });
+  builder: (yargs) => {
+    yargs
+      .option('chains', chainTargetsCommandOption)
+      .option('environment', environmentCommandOption)
+      .option(
+        'out',
+        outputFileCommandOption(
+          './configs/agent-config.json',
+          false,
+          'The path to output an agent config JSON file.',
+        ),
+      )
+      .check((argv) => {
+        if (!argv.chains && !argv.environment) {
+          throw new Error(
+            'Either --chains or --environment must be specified.',
+          );
+        }
+        return true;
+      })
+      .check((argv) => {
+        if (argv.chains && argv.environment) {
+          throw new Error(
+            '--chains and --environment cannot be specified together.',
+          );
+        }
+        return true;
+      });
 
-  //   return yargs as any as Argv<{
-  //     chains?: string | undefined;
-  //     environment?: string | undefined;
-  //     out: string;
-  //   }>;
-  // },
-  handler: async ({ context, chains, environment, out }) => {
+    return yargs as Argv<{
```

Comment:
Doing a type assertion hear. This was the best way that I found to validate the --chains and --environment, ensuring that they are mutually exclusive.

Our `CommandModuleWithContext` type, which we need to pass context the the handle method, also means that the builder needs to return the context when defined as a callback in this way. If anyone has tackled type issues related to yargs,  open to discussing something different